src/main/java/com/vestmark/bitbucket/plugin/MergeConflictDetectorService.java

```diff
@@ -0,0 +1,211 @@
+        //branches.stream()
+        Iterable<Branch> branches = bc.getAllBranchesByType(toBranchType);
+        StreamSupport.stream(branches.spliterator(), true)
+              .filter(b -> mcd.isRelated(b))
+              .filter(b -> mcd.isUpstreamBranch(b))
+              .sorted(branchComparator)
+        //     .forEachOrdered(b -> dryRunMerge(mcd, b));
+              .forEach(b -> dryRunMerge(mcd, b));
+      }
+      dryRunMerge(mcd, refService.getDefaultBranch(mcd.getToRepo())); // Always merge to default.
+      checkForAutoMergeFailure(mcd);
+      return Response.ok(mcd.getMergeResultsModelList()).build();
+    }
+
+    private void checkForAutoMergeFailure(MergeConflictDetector mcd)
+    {
+      try {
+        PullRequestSearchRequest prsReq = new PullRequestSearchRequest.Builder().state(PullRequestState.OPEN)
+          .order(PullRequestOrder.NEWEST)
+          .fromRepositoryId(mcd.getFromRepo().getId())
+          .toRepositoryId(mcd.getToRepo().getId())
+          .build();
+        long totalNPRs = pullRequestService.count(prsReq);
+        int maxNPRs = 100;
+        if (totalNPRs < maxNPRs) {
+          maxNPRs = (int) totalNPRs;
+        }
+        int begIdx = 0;
+        while (begIdx < totalNPRs) {
+          PageRequest pgReq = new PageRequestImpl(begIdx, maxNPRs).buildRestrictedPageRequest(maxNPRs);
+          Page<PullRequest> pPRs = pullRequestService.search(prsReq, pgReq);
+          boolean amfPRExists = pPRs.stream().anyMatch(r -> r.getTitle().equals(AUTO_MERGE_FAIL));
+          if (amfPRExists) {
+            mcd.addResult(
+              refService.getDefaultBranch(mcd.getToRepo()),
+              Collections.emptyList(),
+              Arrays.asList("Please check for " + AUTO_MERGE_FAIL + "!"),
+              Collections.emptyList());
+            return;
+          }
+          begIdx += maxNPRs;
+        }
+      }
+      catch (Exception e) {
+        mcd.addResult(refService.getDefaultBranch(mcd.getToRepo()), null, Arrays.asList(e.getMessage()), null);
+      }
+    }
+
+    public void dryRunMerge(MergeConflictDetector mcd, Branch toBranch)
+    {
+      List<String> files = null;
+      List<GitMergeConflict> mergeConflicts = null;
+      List<String> message = new LinkedList<String>();
+      MergeCommandParameters params = new MergeCommandParameters
+        .Builder()
+        .dryRun(true)
+        .author(mcd.getUser())
+        .fromBranch(mcd.getFromBranchId())
+        .toBranch(toBranch.getId())
+        .message("MergeConflictDetector dry run merge check.")
+        .build();
+      try {
+        Branch result = extendedCmdFactory.merge(mcd.getToRepo(), params).call();
+        // A result from the merge indicates an unsuccessful dry run!
+        if (result != null) {
+          message.add("Merge committed! Commit ID: " + result.getLatestCommit());
+        }
+      } catch (Exception e) {
```

Comment:
catch MergeException, implement your logic therein, then catch Exception and add the generic handling